core/schedule.go

```diff
@@ -27,9 +27,15 @@ import (
 	"github.com/intelsdi-x/snap/pkg/schedule"
 )
 
+// Schedule defines a scheduler.
+//
+// swagger:model Schedule
 type Schedule struct {
-	Type           string     `json:"type,omitempty"`
-	Interval       string     `json:"interval,omitempty"`
+	// required: true
+	// enum: simple, windowed, streaming, cron
+	Type string `json:"type"`
```

Comment:
of course, for the consistent reason or whatever, if team insists to leave it as not required. I will do that but adding a default value.